scripts/core/utils.tsx

```diff
@@ -53,11 +53,16 @@ function requestListener() {
 if (language === 'en') {
     applyTranslations(DEFAULT_ENGLISH_TRANSLATIONS);
 } else {
-    const req = new XMLHttpRequest();
-
-    req.addEventListener('load', requestListener);
-    req.open('GET', filename, false);
-    req.send();
+    fetch(filename).then((response) => {
```

Comment:
By allowing it to be async, some code might execute sooner and call `gettext` before translations are loaded.

See if you can keep the old blocking version.

In the long run we should extract translation loading out of the main bundle in order to use gettext synchronously at the top level in the main bundle.